### ISSUE TITLE
ReciPy

### ISSUE THREAD

**Issue by romulogoncalves** [NLeSC/TEAM2018#41]
@jvdzwaan could you elaborate a bit on what ReciPy can do and what we could do during the Sprint?
Who else has worked with ReciPy?
What is the best label for it `Software dev` or `Soft/Meth Paper`?

**Comment by jvdzwaan** [comment]
Recipy is a Python package that automatically logs provenance information of Python scripts and notebooks. Github repo is https://github.com/recipy/recipy

The main feature is that recipy automatically tracks which files you open and which files you save. This helps people remembering/reconstructing what they did/how certain output was produced.

The use case is python scripts/notebooks, it doesn't work for workflows, or anything more complicated than scripts.

As far as I know recipy is not used in any nlesc projects. For eWaterCycle II, I have been working on some features:

- logging inputs and outputs netcdf/xarray/iris
- Jupyter notebook support
- exporting recipy provenance to PROV standard

These features are more or less finished. However, it seems that eWaterCycle II has evolved to a point where I'm not sure recipy can be used/is really useful. Although we are not using it in projects, I think the software is promising, especially for less tech-savvy fields (i.e., field where researchers write python scripts to analyze results).

During the sprint I/we could either work on features or on a software publication. There are lots of open issues: https://github.com/recipy/recipy/issues
In my free time I have been working on setting up documentation (the first version is nearing completion). A sneak preview can be found at https://recipy.readthedocs.io/en/better-docs/ There also is a draft paper. The most difficult part is describing (potential) impact.

**Comment by romulogoncalves** [comment]
@jvdzwaan  is there project hours for this work? If yes, which one?

**Comment by jvdzwaan** [comment]
I don't think so. The recent work on recipy was done within eWaterCycle II, but as I mentioned before, I'm not sure recipy is going to be there used after all.
